utils/include/vector.h

```diff
@@ -0,0 +1,65 @@
+#ifndef VECTOR_H
+#define VECTOR_H
+
+#include <stddef.h>
+
+typedef struct {
+    size_t capacity;
+    size_t used;
+    size_t item_size;
+    char* contents;
+} vector_t;
+
+/*!
+ * @brief     Initializes a new vector of size 0
+ *
+ * @param     item_size Size of the items to be stored in the vector in bytes,
```

Comment:
The direction needs to be specified on all params, right?

Reply:
I'm not sure... I know its legal to not specify.  I think its only necessary when the param is a pointer, as passing a ``size_t`` will never be an out param.  Its also more important in Javadoc, which is where the style comes from, because all non built-in types are references.